### PR TITLE
Action/2317199 fetch entity by

### PR DESCRIPTION
**Pull request by milkovsky**
Added todo to get entity types list in future.
Changed entity_type to entity_type_id for now.

### CODE REVIEW

File: src/Plugin/Action/FetchEntityById.php [hunk 1]
```diff
@@ -37,6 +37,7 @@
  * )
  *
  * @todo: Add access callback information from Drupal 7.
+ * @todo: port for rules_entity_action_type_options.
  */
 class FetchEntityById extends RulesActionBase implements ContainerFactoryPluginInterface {
```
Comment (klausi) [inline src/Plugin/Action/FetchEntityById.php]: The class should be renamed to EntityFetchById, and also the plugin name.

File: src/Plugin/Action/FetchEntityById.php [hunk 2]
```diff
@@ -87,9 +88,9 @@ public function summary() {
    * {@inheritdoc}
    */
   public function execute() {
-    $entity_type = $this->getContextValue('entity_type');
+    $entity_type_id = $this->getContextValue('entity_type_id');
     $entity_id = $this->getContextValue('entity_id');
-    $storage = $this->entityManager->getStorage($entity_type);
+    $storage = $this->entityManager->getStorage($entity_type_id);
     $entity = $storage->load($entity_id);
     $this->setProvidedValue('entity', $entity);
```
Comment (klausi) [inline src/Plugin/Action/FetchEntityById.php:98]: We should have a todo here for the refinement of the provided context definition for 'entity'. Example: if the loaded entity is a node then the provided context definition should use the type node. We don't have an API for that yet, though.

**Comment by klausi** [comment]
There are conflicts, so you need to merge in the 8.x-3.x changes.

File: src/Plugin/Action/EntityFetchById.php [hunk 3]
```diff
@@ -13,14 +13,14 @@
...
 
 /**
- * Provides a 'Fetch entity by id' action.
+ * Provides a 'Entity fetch by id' action.
  *
  * @Action(
- *   id = "rules_fetch_entity_by_id",
- *   label = @Translation("Fetch entity by id"),
+ *   id = "rules_entity_fetch_by_id",
+ *   label = @Translation("Entity fetch by id"),
```
Comment (fago) [inline src/Plugin/Action/EntityFetchById.php]: label should not change, only the class.

File: tests/src/Integration/Action/EntityFetchByIdTest.php [hunk 4]
```diff
@@ -13,7 +13,7 @@
  * @coversDefaultClass \Drupal\rules\Plugin\Action\FetchEntityById
```
Comment (klausi) [inline tests/src/Integration/Action/EntityFetchByIdTest.php]: Class name needs to be updated here.

**Comment by klausi** [comment]
Merged, thanks.

**Pull request closed without merging**